### PR TITLE
Fixes lexing multiline strings that contain whitespace

### LINKED ISSUE #10: Exception in multiline string handling
TG/Bee contains the following code:
```
/mob/living/simple_animal/hostile/construct/builder
	name = "Artificer"
	real_name = "Artificer"
	desc = "A bulbous construct dedicated to building and maintaining the Cult of Nar'Sie's armies."
	icon_state = "artificer"
	icon_living = "artificer"
	maxHealth = 50
	health = 50
	response_harm = "viciously beats"
	obj_damage = 60
	melee_damage = 5
	retreat_distance = 10
	minimum_distance = 10 //AI artificers will flee like fuck
	attacktext = "rams"
	environment_smash = ENVIRONMENT_SMASH_WALLS
	attack_sound = 'sound/weapons/punch2.ogg'
	construct_spells = list(/obj/effect/proc_holder/spell/aoe_turf/conjure/wall,
							/obj/effect/proc_holder/spell/aoe_turf/conjure/floor,
							/obj/effect/proc_holder/spell/aoe_turf/conjure/door,
							/obj/effect/proc_holder/spell/aoe_turf/conjure/soulstone,
							/obj/effect/proc_holder/spell/aoe_turf/conjure/construct/lesser,
							/obj/effect/proc_holder/spell/targeted/projectile/magic_missile/lesser)
	playstyle_string = "<b>You are an Artificer. You are incredibly weak and fragile, but you are able to construct fortifications, \

						use magic missile, repair allied constructs, shades, and yourself (by clicking on them), \
						<i>and, most important of all,</i> create new constructs by producing soulstones to capture souls, \
						and shells to place those soulstones into.</b>"
						
```
Notice the whitespace line in `playstyle_string`. BYOND ignores whitespace lines but OpenDream throws `"Expected '"' to end string"`

### PR DESCRIPTION
**Pull request by ike709**
**Disclaimer: I am not 100% sure this works as intended**

String parsing is still a bit above my paygrade, but I _think_ this fixes #10 

The main thing I'm iffy on is whether or not `textBuilder.Append('\t');` is needed.

### CODE REVIEW

**Review by wixoaGit — CHANGES_REQUESTED** [review]
This wouldn't give the desired result

Take the following code (there are two tabs on the second line):
```
var/str = "A\
		
	B"
```

This ends up the same as `var/str = "AB"`. The solution to this is likely to skip any whitespace/newlines after a line ending in a backslash. This would be done after [this line](https://github.com/wixoaGit/OpenDream/blob/e76c575576a59dffb48981f160f7ecc76d742c40/OpenDreamShared/Compiler/DMPreprocessor/DMPreprocessorLexer.cs#L187).

**Comment by ike709** [comment]
Done?

**Comment by ike709** [comment]
Done?

File: OpenDreamShared/Compiler/DMPreprocessor/DMPreprocessorLexer.cs [hunk 1]
```diff
@@ -194,6 +194,11 @@ class DMPreprocessorLexer : TextLexer {
                 if (_source[_currentPosition] == '\n') { //Skip a newline if it comes after a backslash
                     base.Advance();
 
+                    while (current == ' ' || current == '\t' || current == '\n')
+                    {
+                        current = Advance();
+                    }
+                    
                     current = Advance();
```
Comment (wixoaGit) [inline OpenDreamShared/Compiler/DMPreprocessor/DMPreprocessorLexer.cs:202]: This line should be removed, it ends up skipping the first non-whitespace character

**Review by wixoaGit — CHANGES_REQUESTED** [review]

File: OpenDreamShared/Compiler/DMPreprocessor/DMPreprocessorLexer.cs [hunk 2]
```diff
@@ -194,6 +194,11 @@ class DMPreprocessorLexer : TextLexer {
                 if (_source[_currentPosition] == '\n') { //Skip a newline if it comes after a backslash
                     base.Advance();
 
+                    while (current == ' ' || current == '\t' || current == '\n')
```
[the remark this answers is not in the document]
Comment (ike709) [reply OpenDreamShared/Compiler/DMPreprocessor/DMPreprocessorLexer.cs:197]: And from this I have discerned that I was testing against a version of the fork that I'd removed multiline whitespaces from.

**Comment by ike709** [comment]
Woopsie, this was tested against actual nonedited BeeStation now and it seems to function as intended.

**Review by wixoaGit — APPROVED** [review]
Looks good, seems to work well

**Pull request merged**